src/minErr.js

```diff
@@ -34,19 +34,32 @@ function minErr(module, ErrorConstructor) {
   ErrorConstructor = ErrorConstructor || Error;
   return function() {
     var SKIP_INDEXES = 2;
+    var DEFAULT_MAX_DEPTH = 3;
 
     var templateArgs = arguments,
+      argsDebuggedStringIndexes = [],
       code = templateArgs[0],
       message = '[' + (module ? module + ':' : '') + code + '] ',
       template = templateArgs[1],
       paramPrefix, i;
 
-    message += template.replace(/\{\d+\}/g, function(match) {
-      var index = +match.slice(1, -1),
-        shiftedIndex = index + SKIP_INDEXES;
+    function getArgumentAndConvertToDebugString(index, maxDepth) {
```

Comment:
It doesn't make much sense to make `maxDepth` configurable per argument (or even per call).
What we need is for the user to be able to configure it globally. `minErr` is private, so it is Angular that calls `minErr`, not the user and having to configure `maxDepth` for each argument is a burden, not a feature.

The first things that come to mind for making `maxDepth` globally configurable are:

1. Adding a new helper function on `angular`. E.g. `angular.setObjectMaxDepthInErrors()` (the name could be better - but you get the idea :smiley:)
2. Adding a new service/provider for configuring `maxDepth` (but it might miss some early errors, before bootstrapping the app).
3. Adding it as a property in the app's bootstrap config (for manual bootstrapping). This would be the best if it weren't for automatically bootstrapped apps (via `ngApp`). Now we need a `ng-object-max-depth-in-errors` directive (similar to `ng-strict-di`), which is not cool.

I think I prefer (1). Seems to be the simplest to implement and having the less shortcomings.